### PR TITLE
add bigquery connection with examples

### PR DESCRIPTION
**Pull request by hcliff**
https://github.com/terraform-providers/terraform-provider-google/issues/5530

notes:
1) it's unclear on if this is ga or not to me, the api url says beta, so i went with beta.
2) I had to declare `google-beta` as the provider for everything in the examples, or the terraform go modules wouldn't build, what's the desired way to do this?
3) name vs id was not clear to me, i mapped the properties directly from the api, but this leads to the oddly reading: 
`instance_id = google_sql_database_instance.instance.connection_name`
per the docs, `connection_id` is optional (left blank a default is generated for you), which seems different from other products
4) "DATABASE_TYPE_UNSPECIFIED" when set maps to an undefined value on response (the `type` property is not present in the response) - I couldn't find any code to cargo-cult here, suggestions?

```release-note:new-resource
`google_bigquery_connection`
```

### CODE REVIEW

**Comment by hcliff** [comment]
@danawillow while https://github.com/GoogleCloudPlatform/magic-modules/pull/3427 is inflight is is possible to get a review or reviewer? :) I'm not sure what's breaking the build because i can't see it

**Comment by danawillow** [comment]
requesting nathan as per https://github.com/GoogleCloudPlatform/magic-modules/pull/3424#issuecomment-619280576

File: third_party/terraform/tests/resource_bigquery_connection_test.go.erb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<% autogen_exception -%>
```
Comment (ndmckinley) [inline third_party/terraform/tests/resource_bigquery_connection_test.go.erb:1]: Why does this file need the exception?

**Comment by ndmckinley** [comment]
Ah, it's the conflict, of course, that makes sense.  Please rebase to head!

File: products/bigqueryconnection/api.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        name: 'friendlyName'
+        description: A descriptive name for the connection
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: A descriptive description for the connection
+      - !ruby/object:Api::Type::Boolean
+        name: 'hasCredential' 
+        output: true
+        description: |
+          Output only. True if the connection has credential assigned.
```
[the remark this answers is not in the document]
Comment (hcliff) [reply products/bigqueryconnection/api.yaml:76]: 👍

File: templates/terraform/examples/bigquery_connection_basic.tf.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+resource "google_sql_database" "db" {
+    provider = google-beta
+    instance = google_sql_database_instance.instance.name
+    name     = "db"
+}
+
+resource "google_bigquery_connection" "<%= ctx[:primary_resource_id] %>" {
+    provider      = google-beta
+    friendly_name = "👋"
```
[the remark this answers is not in the document]
Comment (hcliff) [reply templates/terraform/examples/bigquery_connection_basic.tf.erb:19]: thanks!

**Comment by ndmckinley** [comment]
Cool, this looks good to me and I'll go ahead and run the integration tests.

**Comment by ndmckinley** [comment]
I tweaked the comment to make the changelog checker happy.  :)

File: third_party/terraform/tests/resource_bigquery_connection_test.go.erb (hunk 1 above)
Comment (hcliff) [reply third_party/terraform/tests/resource_bigquery_connection_test.go.erb:1]: blindly cargoculted, all the .erb test files seem to have it at the top, thoughts?

File: products/bigqueryconnection/api.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        output: true
+      - !ruby/object:Api::Type::String
+        name: connection_id
+        description: |
+          Optional connection id that should be assigned to the created connection. 
+        required: false
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'location'
```
[the remark this answers is not in the document]
Comment (hcliff) [reply products/bigqueryconnection/api.yaml:57]: 👍 I'd rather not put an exhaustive list as it's liable to change as the products wider released, but I've beefed it up to be >= the docs on https://cloud.google.com/bigquery/docs/reference/bigqueryconnection/rest/v1beta1/projects.locations.connections#Connection

**Comment by hcliff** [comment]
thanks @ndmckinley and @danawillow :)

**Comment by preston-hf** [comment]
Apologies if I'm missing something, but shouldn't the credential be able to be provided by a `credential` block or something allowing you to set the username and password? It doesn't seem like this is usable without that, and ideally the test could catch that?

**Comment by ndmckinley** [comment]
Doesn't seem that way: https://cloud.google.com/bigquery/docs/reference/bigqueryconnection/rest/v1beta1/projects.locations.connections#Connection - is this not usable for your use-case as-is?

**Comment by preston-hf** [comment]
I want to setup a connection in BigQuery to Postgres Cloud SQL. I believe that requires a username and password to be specified in a credential, which doesn't seem possible. So no, this isn't usable unless there is a way to have it connect to Cloud SQL without specifying credentials.

**Comment by ndmckinley** [comment]
I don't think it does require a username and password in the connection - I believe that's in the connection string.

This enables these features: https://cloud.google.com/bigquery/docs/cloud-sql-federated-queries.

**Comment by preston-hf** [comment]
In our experience it doesn't work. Connections require a user and users require a password. How else would you control what data is exposed to BigQuery users. We still feel that this is a missing resource. Unless you have tried it and it works, I think this implementation is incomplete.

**Comment by michellen-hf** [comment]
Specifically - the error you get is:
```
Missing credentials on Cloud SQL connection: <connection_name>
```
That's for a `POSTGRES` type connection, but based on the [documentation](https://cloud.google.com/bigquery/docs/cloud-sql-federated-queries#setting_up_database_connections) - It seems true for mysql as well. Additionally the documentation implies there is no way to include credentials in the sql.
> To avoid writing database credentials as cleartext in a federated query, you need to first create a database connection resource per database in BigQuery and then reference the connection resource in your federated query.

**Comment by ndmckinley** [comment]
Aha, the documentation is out of date!  That field does exist, its structure is 

```
{
  ...
  "credential": {
    "username": string,
    "password": string,
  }
}
```

The documentation will be updated soon, but I can go ahead and fix this now.

**Pull request merged**